### PR TITLE
Edit feature

### PR DESCRIPTION
**Pull request by Jefferson111**
- Add tests
- Add edit feature
   - EditorManager, EditorCommand, EditCommand, EditorParser, Selectors
   - SidePanel, PointCard for Ui: Java + FXML + CSS
- Remove Todo Operations
   - Todo is now exclusive to Itinerary
- ParserUtil minor changes
   - Delete getHolidayIndex() and uses the below methods
   - Change getFirstIndex() to get first index
   - Change getSecondIndex() to get second index
   - There are still a lot of unnecessary code lying around in ParserUtil
- Parser
   - Extract and break down some parsers into more parser classes
   - Still need Parser interface that implements method parse()
- Resolve merge conflicts in Javadoc comments

### CODE REVIEW

File: src/main/java/duke/logic/commands/DeleteCommand.java [hunk 1]
```diff
@@ -30,9 +30,9 @@ public DeleteCommand(int index) {
     @Override
     public CommandResultText execute(Model model) throws DukeException {
         try {
-            Task task = model.getTasks().remove(index);
+            Event event = model.getEvents().remove(index);
             model.save();
-            return new CommandResultText(MESSAGE_DELETE + task);
+            return new CommandResultText(MESSAGE_DELETE + event);
         } catch (IndexOutOfBoundsException e) {
             throw new DukeException(Messages.ERROR_INDEX_OUT_OF_BOUNDS);
```
Comment (Inno97) [inline src/main/java/duke/logic/commands/DeleteCommand.java]: Perhaps we need a specific exception for index out of bounds.
We should be also able to know what was the query that caused the exception too.

File: src/test/java/duke/logic/EditorManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        result = EditorManager.edit(KeyCode.UP);
+        assertTrue(result.isReady());
+        result = EditorManager.edit(KeyCode.ENTER);
+        assertTrue(result.isReady());
+        result = EditorManager.edit(KeyCode.ENTER);
+        assertTrue(result.isReady());
+        result = EditorManager.edit(KeyCode.ESCAPE);
+        assertTrue(result.isReady());
+        result = EditorManager.edit(KeyCode.ESCAPE);
+        assertTrue(result.isReady());
```
Comment (Inno97) [inline src/test/java/duke/logic/EditorManagerTest.java]: Is it possible to do negative testing?

Comment (Jefferson111) [reply src/test/java/duke/logic/EditorManagerTest.java]: I don't think so, for this particular method, I'm testing how the EditorManager will handle keyEvents, and under no circumstances the Class should fail, according to the design.

File: src/main/java/duke/logic/commands/DeleteCommand.java (hunk 1 above)
Comment (Jefferson111) [reply src/main/java/duke/logic/commands/DeleteCommand.java]: 👌

**Pull request merged**